src/main/java/tdl/client/queue/QueueBasedImplementationRunner.java

```diff
@@ -1,29 +1,214 @@
+                customConfig.getString("sqs.secretKey")
+        );
+
+        try {
+            messageRequestQueue = createQueueRequest(customConfig.getString("sqs.requestQueueName"));
+            messageRequestQueueUrl = createQueue(messageRequestQueue);
+            messageResponse = createQueueRequest(customConfig.getString("sqs.responseQueueName"));
+            messageResponseQueueUrl = createQueue(messageResponse);
+        } catch (Exception ex) {
+            logToConsole("        QueueBasedImplementationRunner [error]");
+            String message = "There was a problem creating the queue runner";
+            LOGGER.error(message, ex);
+            audit.logException(message, ex);
+        }
+
+        receiveRequestQueueMessage = createReceiveRequest(messageRequestQueueUrl);
+
+        mapper = new ObjectMapper();
+
+        GsonBuilder gsonBuilder = new GsonBuilder();
+        gsonBuilder.registerTypeAdapter(Double.class,
+                (JsonSerializer<Double>) (src, typeOfSrc, context) -> {
+                    Integer value = (int) Math.round(src);
+                    return new JsonPrimitive(value);
+                });
+
+        gson = gsonBuilder.serializeNulls().create();
+
+        batchOfMessagesToDeleteRequest = new DeleteMessageBatchRequest().withQueueUrl(messageResponseQueueUrl);
+        deleteMessageBatchRequestEntries = batchOfMessagesToDeleteRequest.getEntries();
+        
+        logToConsole("        QueueBasedImplementationRunner creation [end]");
+    }
+
+    private ReceiveMessageRequest createReceiveRequest(String queueUrl) {
+        return new ReceiveMessageRequest()
+                .withMaxNumberOfMessages(MAX_NUMBER_OF_MESSAGES)
+                .withQueueUrl(queueUrl)
+                .withWaitTimeSeconds(MAX_AWS_WAIT)
+                .withMessageAttributeNames(Arrays.asList(ATTRIBUTE_EVENT_NAME, ATTRIBUTE_EVENT_VERSION));
+    }
+
+    public List<String> getResponseQueueMessages() throws IOException {
+        List<String> results = new ArrayList<>();
+
+        ReceiveMessageRequest receiveMessageRequest = createReceiveRequest(messageResponseQueueUrl);
+        ReceiveMessageResult batch;
+        final List<Message> messages = new ArrayList<>();
+        do {
+            batch = client.receiveMessage(receiveMessageRequest);
+            messages.addAll(batch.getMessages());
+        } while (batch.getMessages().size() > 0);
+
+        for (int index = messages.size() - 1; index >= 0; index--) {
+            Message message = messages.get(index);
+            JsonNode jsonNode = mapper.readValue(message.getBody(), JsonNode.class);
+            String payload = jsonNode.get("payload").asText();
+            JsonRpcResponse jsonRpcResponse = gson.fromJson(payload, JsonRpcResponse.class);
+            Response response = jsonRpcResponse.toResponse();
+
+            //TODO: implementation needs replacing, for some reason GSon is converting all int values into double
```

Comment:
@julianghionoiu Can we look into this, I have build a GSon builder up above, and also passed the needed parameters, it still seem to convert integers into floats (single/double) and this is causing the tests to fail, the scaffolding here is just to temporarily overcome that.